WinAppDriver/CommandHandlers/NewSessionCommandHandler.cs

```diff
@@ -61,15 +63,57 @@ public override Response Execute(CommandEnvironment environment, Dictionary<stri
             var desiredCapabilities = JObject.Parse(parameters["desiredCapabilities"]?.ToString() ?? "{}").ToObject<Dictionary<string, object>>();
             //parameters["desiredCapabilities"] as Dictionary<string, string> ?? new Dictionary<string, string>();
 
-            if (!desiredCapabilities.TryGetValue("processName", out var processName))
+            if (!desiredCapabilities.TryGetValue("processName", out var processName) 
+                & !desiredCapabilities.TryGetValue("exePath", out var exePath))
             {
-                return Response.CreateMissingParametersResponse("processName");
+                return Response.CreateMissingParametersResponse("processName or exePath");
             }
 
-            var process = Process.GetProcessesByName(processName.ToString()).FirstOrDefault();
-            if (process == null)
+
+            Process process = null;
+            if (processName != null)
             {
-                return Response.CreateErrorResponse(-1, $"Cannot attach to process '{processName}', no such process found.");
+                
+                process = Process.GetProcessesByName(processName.ToString()).FirstOrDefault();
+
+                if (process == null)
+                {
+                    var processes = Process.GetProcesses();
+                    process = processes.Where(x =>
```

Comment:
Vladyslav, can you please explain this part? When process name is specified I want that process, not process that contains that name, but a process whose name exactly matches the desired capability value.

Reply:
Sorry for missing comments. In further I'll write comments. That piece of code I decided to make after long searching of calculator process name. I found it under debugging. If the specified capability value exactly matches the desired process name, the process variable won't be null(line 77). Hence searching by part of the name won't be executed.

But I understand that it's might be a strange solution. It's the first thing that came to mind.

Also, I think need to add search by the path to process execution file. If an application already runs and in the capability set the only path to executable my code creates a new process. 

What do you think about these changes?
Do you think I need to remove the part name search or change its implementation?

Reply:
Please change the implementation so it will use regular expression. The idea to launch the program is good. The desired capabilities collection can be extended with `mode` which will tell either to attach or start new process.
This tool https://docs.microsoft.com/en-us/windows/desktop/winauto/ui-automation-verify will help you in your work (find the process name, element name, etc).

Reply:
I am using Inspect.exe, but both of them show me a localized process name f.e. "Калькулятор".